axiom/nr_php_packages.c

```diff
@@ -0,0 +1,142 @@
+/*
+ * Copyright 2022 New Relic Corporation. All rights reserved.
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#include <math.h>
+#include <stdlib.h>
+#include <stdio.h>
+
+#include "nr_axiom.h"
+#include "nr_limits.h"
+#include "nr_php_packages.h"
+#include "util_memory.h"
+#include "util_minmax_heap.h"
+#include "util_vector.h"
+#include "util_hashmap.h"
+#include "util_hashmap_private.h"
+#include "util_strings.h"
+
+nr_php_package_t* nr_php_package_create(char* name, char* version) {
+  nr_php_package_t* p = NULL;
+
+  if (name == NULL || version == NULL) {
+    return NULL;
+  }
+
+  p = (nr_php_package_t*)nr_malloc(sizeof(nr_php_package_t));
+
+  p->package_name = nr_strdup(name);
+  p->package_version = nr_strdup(version);
+
+  return p;
+}
+
+void nr_php_package_destroy(nr_php_package_t* p) {
+  if (p != NULL) {
```

Comment:
To avoid a situation of accessing an invalid memory by referencing a pointer to memory that has been freed, I suggest `nr_php_package_destroy` to set `p` to `NULL`:
```suggestion
void nr_php_package_destroy(nr_php_package_t** ptr) {
  nr_php_package_t* p;

  if ((NULL == ptr) || (NULL == *ptr)) {
    return;
  }

  p = *ptr;

  if (p != NULL) {
```